### PR TITLE
Allow importing externally allocated memory into an allocator

### LINKED ISSUE #60: Allow adding MemoryBlocks manually to GpuAllocator
When importing external memory I need to use facilities outside of a memory device and the gpu allocator to create the device memory. Generally via extensions like VK_EXT_external_memory_dmabuf and etc.

Once I have imported the memory I would like a way to have the allocator track the external memory I have imported so that cleanup can manage freeing the allocations from external memory.

I guess a way to know how many allocations are left would also be needed to use said api safely.

### PR DESCRIPTION
**Pull request by i509VCB**
~~There is an open question that needs to be addressed regarding this pull request whether the imported memory block belongs to some heap before merging.~~

Fixes #60

### CODE REVIEW

File: gpu-alloc/src/allocator.rs [hunk 1]
```diff
@@ -397,6 +397,71 @@ where
...
+    pub unsafe fn import_memory(
+        &mut self,
+        memory: M,
+        memory_type: u32,
+        props: MemoryPropertyFlags,
+        offset: u64,
+        size: u64,
+    ) -> MemoryBlock<M> {
+        // TODO: Before merge:
+        // - Should this memory belong to a heap?
```
Comment (zakarumych) [inline gpu-alloc/src/allocator.rs:433]: If it was allocated from a heap

**Review by zakarumych — COMMENTED** [review]
This looks good.

I think we can also provide methods to import external memory to the implementation-specific crates like `gpu-alloc-ash` and `gpu-alloc-erupt`.
Then user would use one of the provided methods if they fit user's usecase and then call this method to import it into `GpuAllocator`.

File: gpu-alloc/src/allocator.rs (hunk 1 above)
Comment (i509VCB) [reply gpu-alloc/src/allocator.rs:433]: What type of memory would be classified as belonging to a heap?

Comment (zakarumych) [reply gpu-alloc/src/allocator.rs:433]: When memory object is allocated, it is allocated with a memory type, bounded to a memory heap.
When importing memory object, if it is associated with a heap, heap usage should be updated.

**Comment by i509VCB** [comment]
@zakarumych I hope you don't mind a ping that I have resolved the issues with this pull request.

**Review by zakarumych — APPROVED** [review]
This looks good.
Thank you!

**Pull request merged**